homeassistant/components/ring/switch.py

```diff
@@ -0,0 +1,115 @@
+"""This component provides HA switch support for Ring Door Bell/Chimes."""
+from datetime import timedelta
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+
+from . import DATA_RING
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=10)
+
+SIREN_ICON = 'mdi:alarm-bell'
+LIGHT_ICON = 'mdi:track-light'
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up a sensor for a Ring device."""
+    ring = hass.data[DATA_RING]
+    logging.warning('setting up')
+    switches = []
+
+    for device in ring.stickup_cams:  # ring.stickup_cams is doing I/O
+        if device.has_capability('siren'):
+            switches.append(SirenSwitch(hass, device))
+        if device.has_capability('light'):
+            switches.append(LightSwitch(hass, device))
+
+    add_entities(switches, True)
+
+
+class BaseRingSwitch(SwitchDevice):
+    """Represents a switch for controlling an aspect of a ring device"""
+    def __init__(self, hass, device, device_type):
+        self._device = device
+        self._device_type = device_type
+        self._unique_id = '{}-{}'.format(self._device.id, self._device_type)
+
+    @property
+    def name(self):
+        """Name of the device."""
+        return '{} {}'.format(self._device.name, self._device_type)
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._unique_id
+
+    def update(self):
+        """Get the siren status from ring"""
+        self._device.update()
+
+
+class SirenSwitch(BaseRingSwitch):
+    """Used to turn the ring cameras siren on and off"""
+    def __init__(self, hass, device):
+        BaseRingSwitch.__init__(self, hass, device, 'siren')
+        self._on = False
+
+    @property
+    def is_on(self):
+        """If the switch is currently on or off."""
+        return self._on
+
+    def turn_on(self, **kwargs):
+        """Turn the siren on for 30 seconds."""
+        self._on = True
```

Comment:
How fast is the feedback from calling `update`? The `update` method is called at the end of the service call after the actuator method call, for polling entities. If the feedback is fast enough, < 2 seconds, from the device we don't need to change state holding instance attribute here.

Reply:
I'm going to fix this as a second pull request. I'd like to get this released, and the fix for this will impact the rest of the ring.